File: 0901-1000/0991-Broken Calculator/0991-Broken Calculator.py
class Solution:
    def brokenCalc(self, X: int, Y: int) -> int:
        num = 0
        base = 1
        while X * base < Y:
            num += 1
            base <<= 1
        diff = X * base - Y
        while diff:
            num += diff // base
            diff %= base
            base >>= 1
        return num

class Solution2:
    def brokenCalc(self, X: int, Y: int) -> int:
        num = 0
        while X < Y:
            if Y & 1:
                Y += 1
            else:
                Y >>= 1
            num += 1

        return num + X - Y

File: 0901-1000/0991-Broken Calculator/0991-Broken Calculator.cpp
class Solution {
public:
    int brokenCalc(int X, int Y) {
        int num = 0;
        while (X < Y) {
            if (Y & 1) {
                ++Y;
            }
            else {
                Y >>= 1;
            }
            ++num;
        }
        
        return num + X - Y;
    }
};
